osm-fudge/__main__.py

```diff
@@ -0,0 +1,27 @@
+import bk_tree
+import sys
+
+import levenshtein
+import jaccard
+
+distance_metrics = {
+	'levenshtein': levenshtein.find_levenshtein_distance,
+	'jaccard': jaccard.find_jaccard_similarity
+}
+
+with open(sys.argv[1], 'r') as file:
+	initial_word = file.readline()
+	metrics = sys.argv[2]
+	if metrics not in distance_metrics.keys():
+		sys.exit("Allowed metrics are {metrics}".format(metrics=list(distance_metrics.keys())))
+	options = {'n' : 2}
+	tree_obj = bk_tree.BKTree(initial_word, distance_metrics[metrics], options)
+
+	for line in file: 
+		if len(line) > 1:
+			tree_obj.insert(line.strip(), tree_obj.tree)
+
+	results = []
+	tree_obj.lookup('parking', tree_obj.tree, 8, results)
```

Comment:
Here is an example where we break the bktree's abstraction: the user has to pass `tree_obj.tree` and therefore has to know about our implementation details. The user's code now is tightly coupled with our implementation and we can never change it.

But what the user only cares about it:
- I have a bktree object
- I want to lookup an item (doesn't even have to be a string)
- I want either up to n results, or results up to a max distance of d (probably good to provide both in the API)

Maybe something like

```python
def lookup(self, item, max_results=None, max_distance=None)
```

?